example/ConquerClub.Domain/Commands/Advance.cs

```diff
@@ -0,0 +1,7 @@
+﻿using GameId = Qowaiv.Identifiers.Id<ConquerClub.Domain.ForGame>;
+
+namespace ConquerClub.Domain.Commands
+{
+    public record Advance(Army To, GameId Game, int ExpectedVersion) 
```

Comment:
Honestly, I would not mind too much in this case. Especially as the domain is so well established that there is probably little reason for it to change within these classes. But normally my primary argument would be that this creates a strong dependency between the internals of the domain and the commands and events. So if one is forced to change, this in turn would force other parts to change with it. In the case of events, I might imagine that this might also give issues with stored/serialized events.

Reply:
I would argue that Army is a way to bound the size of the army, to its owner. Very practical for the domain and communication. If the requirements of the domain change, I would argue, that in this case, this Army representation should stay unchanged, and only the Army as Value Object or Entity of the domain should.